### PR TITLE
Eliminates most of the Rollback code, and things interacting with it

### PR DESCRIPTION
**Pull request by zwoop**
The ultimate goal here is to make etc/ configurations read-only, such that we don't create versioned files, or try to write updates out to the configurations. This is a complex patch, I left the intermediary steps intact in this PR, for easier review, and easier debugging if I break something. There's a lot of cross module interaction here.

Note that there likely are more pieces in FileManager that can be optimized, but this is at least a good start I think.

### CODE REVIEW

**Comment by zwoop** [comment]
[approve ci autest]

File: mgmt/Rollback.cc [hunk 1]
```diff
@@ -34,12 +34,6 @@
 #include "FileManager.h"
```
Comment (bryancall) [inline mgmt/Rollback.cc]: Why do we still have this file?

Comment (zwoop) [reply mgmt/Rollback.cc]: You mean Rollback? Turns out the rollback object was both for rollbacks as well as change management. Change management is still there. I can rename this file if preferable.

**Comment by zwoop** [comment]
I've renamed Rollback.cc -> ConfigManager.cc, and the Rollback class is now named ConfigManager. I think it might be reasonable to incorporate all of this into FileManager, but this is already getting unwieldy as is (what started as a simple goal of making .../etc read-only, is now a big undertaking :).

**Comment by zwoop** [comment]
[approve ci autest]

**Pull request merged**